### ISSUE TITLE
Bulk Change Flight Location

### ISSUE THREAD

**Issue by markmacpherson** [cadets-ca/ets#36]
In Central Region, each flight changes airports every week. When they move to a new airport, each pilot needs to have their profile updated individually to move them to the new site. This is upwards of 100 individual actions to move flight B to site B and flight B to site A each week.

The desired functionality is that the pilots at site 'A' in flight 'A' can be moved in bulk to site B using one action.

**Comment by markmacpherson** [comment]
This will also be an issue in Eastern Region, as they also fly in two different airports

**Comment by kirvanp** [comment]
So the problem here is that the app doesn't support any way of tracking which airport is being used. It was built around the central concept of a gliding unit, that being a group of people who typically fly together at a particular airport. On the rare times when a gliding center deployed to another location, the app didn't record that and it was up to the staff to note the different location in the paper Journey Log. You want to expand the app to include units with multiple sites and powered fam flying at any arbitrary site if I understand correctly. The trouble is, adding each airport as a "Unit" leads to problems like the one you described where it is difficult to know what staff fly where if people keep switching between airports. 

Allow me to propose a simpler solution. Keep the units broad, e.g. Eastern CFTC, and don't make each airport a unit. Add a "airport" attribute to the timesheet entity in the sql model. Then, when timesheets are created, have the iPad use it's GPS to record the airport in use. The airport name can then be added to the header of the timesheets when they are exported to HTML and PDF. The iPad already can track the route in use for x-country flights, so you can reuse that code. This solution requires absolutely zero effort on the part of the user the code changes should be much easier, and it would be able to log any known ICAO aerodrome vs. having to hard code the different sites into the app.

**Comment by markmacpherson** [comment]
The units were separated because the two sites operate simultaneously and they didn't want each others records showing up on the other site's iPad throughout the day, if they were connected to the internet and syncing with iCloud. They operate like their own 'unit' (sub-unit maybe?), they just switch airports every week. 

Maybe the sites should be named differently? Currently there is Mountainview CFTC and Mountainview CFTC at Picton. Maybe we could give them generic names like MVCFTC - A and MVCFTC - B and then add in an Airport field like you mentioned, to track where the flying is taking place. This is still a little clunky, but gives them the separation they are looking for. 

Alternatively, we could allow the list of flights to be filtered by Airport, so they could all stay under one 'Unit' (Mountain View CFTC) but then be able to filter out the flights from the other airport. This might be the better way to do it, since it makes reporting a little simpler, and also make it easy to understand where to put a transit flight between the two airports.

**Comment by kirvanp** [comment]
Good point, I forgot about the syncing. The summer camp flights though are independent of the units, though the list of which ones are available is filtered by region. So, for example, I could sign in all 'Bandit' flight people even if I have Peace River CFS selected. There shouldn't therefore be a need to transfer people from one flight to another in bulk. So why not have two units, Mountainview and Picton and sign in the appropriate flight at each one? The app wouldn't sync between them and the timesheets would be labelled as such.

**Comment by markmacpherson** [comment]
That's what I initially thought they should do, but when they go to sign a flight in, it is only signing in the members of X flight who are also part of the Site they are trying to sign them in to. So, instead they are moving people to the site, and then signing in the flight. Seems like maybe we could just tweak the way the "sign in X flight" function works maybe?
